src/components/Search/CompareTable.jsx

```diff
@@ -0,0 +1,151 @@
+        <TableHead>
+          <TableRow>
+            <StyledTableCell />
+            <StyledTableCell
+              align="left"
+              style={{
+                fontWeight: 600,
+              }}
+            >
+              Project
+            </StyledTableCell>
+            <StyledTableCell align="left" style={{ fontWeight: 600 }}>
+              Revision
+            </StyledTableCell>
+            <StyledTableCell align="left" style={{ fontWeight: 600 }}>
+              Author
+            </StyledTableCell>
+            <StyledTableCell align="center" style={{ fontWeight: 600 }}>
+              Commit message
+            </StyledTableCell>
+            <StyledTableCell align="left" style={{ fontWeight: 600 }}>
+              Timestamp
+            </StyledTableCell>
+            <StyledTableCell />
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {revisions.map((row) => (
+            <StyledTableRow key={row.id}>
+              <StyledTableCell component="th" scope="row">
+                <div
+                  style={{
+                    fontWeight: 600,
+                    justifyContent: 'center',
+                    borderRadius: '2.5px',
+                    paddingTop: '5px',
+                    paddingBottom: '5px',
+                    border: '1px solid #d3d3d3',
+                    display: 'flex',
+                    color: '#d3d3d3',
+                    width: '50px',
+                  }}
+                >
+                  {row.type.toUpperCase()}
+                </div>
+              </StyledTableCell>
+              <StyledTableCell component="th" scope="row">
+                {row.projectName}
+              </StyledTableCell>
+              <StyledTableCell align="left">{row.revision}</StyledTableCell>
+              <StyledTableCell align="left">{row.author}</StyledTableCell>
+              <StyledTableCell align="center">
+                {row.commitMessage}
+              </StyledTableCell>
+              <StyledTableCell align="left">{row.timeStamp}</StyledTableCell>
+              <StyledTableCell align="right">
+                <IconButton
+                  onClick={() => store.dispatch(deleteRevision(row.id))}
+                >
+                  <CloseIcon />
+                </IconButton>
+              </StyledTableCell>
+            </StyledTableRow>
+          ))}
+        </TableBody>
+      </Table>
+      <p style={{ marginLeft: 'auto', color: 'grey' }}>
+        {revisions.length > 0 ? 'Maximum 4 revisions.' : 'Add some revisions'}
+      </p>
+    </TableContainer>
+  );
+}
+
+CompareTable.propTypes = {
+  revisions: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number,
+      projectName: PropTypes.string,
+      revision: PropTypes.string,
+      author: PropTypes.string,
+      commitMessage: PropTypes.string,
+      timeStamp: PropTypes.string,
+    }),
+  ).isRequired,
+};
```

Comment:
the structure of this dictionary should match up to the current structure of a revision. See https://github.com/mozilla/perfcompare/blob/master/src/components/Search/SearchResultsListItem.jsx#L38-L56

this is the way the data is returned from the API, and your code should be able to integrate with existing code.

Reply:
The API for performance summary takes a revision ID. the actual comparison is currently done on the front end of Perfherder, but as we've discussed this is not desirable and we want to create a new endpoint for comparison results. I'm assuming this would also take revision IDs? We have not discussed the specifics yet.

The revisions do often have more than one entry. Currently I have helper messages to select the "comment" or commit message from only the last revision in the array (usually the first is the temporary mach commit that includes `try_task_config.json` saying which tests to run)

you make some good points. As I said I already have a helper method to select only the most recent revision, we could later add a helper method to create a new revision item.

Reply:
I'm going to merge the code as is, and will handle in the different patch, the connection between selecting revision and adding to the table havent been done yet, probably best to handle that there.